### PR TITLE
Feature/split load block call alt 580

### PR DESCRIPTION
**Pull request by blackyblack**

### CODE REVIEW

**Comment by black0rwhite** [comment]
Perhaps, loadLeafBlock and loadRootBlock would be more descriptive than loadBlockForward and loadBlockBackward?

While I don't know the intent and context of these changes, I think that loading leaf blocks before the root one is impractical considering the upcoming change towards storing block references as pointers instead of hashes in endorsements. Such a change disallows dangling references and thus loading blocks in the leaves-to-root order will have to happen in 2 passes:
1) instantiate empty blocks
2) populate endorsements and possibly other data

**Comment by Mr-Leshiy** [comment]
@black0rwhite  The intention of this PR is to split the current functionality of the loadBlock and narrowing the scope of the each method.

File: include/veriblock/pop/blockchain/base_block_tree.hpp [hunk 1]
```diff
@@ -157,89 +157,7 @@ struct BaseBlockTree {
    * state.
    */
   virtual bool loadBlock(const stored_index_t& index, ValidationState& state) {
```
Comment (Mr-Leshiy) [inline include/veriblock/pop/blockchain/base_block_tree.hpp:159]: Remove this function, just leave only loadBlockForward loadBlockBackward and make them `public` and `virtual` as with the loadBlock

File: include/veriblock/pop/blockchain/blocktree.hpp [hunk 2]
```diff
@@ -161,6 +162,35 @@ struct BlockTree : public BaseBlockTree<Block> {
...
+
+    VBK_ASSERT(current->pnext.size() == 1);
+    auto* next = *current->pnext.begin();
+
+    // recover chainwork
+    auto chainWork = getBlockProof(next->getHeader());
+    current->chainWork = next->chainWork - chainWork;
+
+    // clear blockOfProofEndorsements inmem field
+    current->clearBlockOfProofEndorsement();
```
Comment (blackyblack) [inline include/veriblock/pop/blockchain/blocktree.hpp:188]: I think it is better to avoid duplicate code by using loadBlockInner.

**Pull request merged**